File: src/main/java/zampano/ai/zampanoreminder/domain/Reminder.java
package zampano.ai.zampanoreminder.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reminders")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reminder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String notes;

    private String url;

    private LocalDateTime remindAt;

    @Column(nullable = false)
    @Builder.Default
    @Enumerated(EnumType.STRING)
    private Priority priority = Priority.NONE;

    @Column(nullable = false)
    @Builder.Default
    private boolean flagged = false;

    @Column(nullable = false)
    @Builder.Default
    private boolean completed = false;

    @Column(nullable = false)
    @Builder.Default
    private int sortOrder = 0;

    @Column(nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(nullable = false)
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "list_id")
    private ReminderList reminderList;

    public void update(String title, String notes, String url, LocalDateTime remindAt, Priority priority, boolean flagged) {
        this.title = title;
        this.notes = notes;
        this.url = url;
        this.remindAt = remindAt;
        this.priority = priority;
        this.flagged = flagged;
        this.updatedAt = LocalDateTime.now();
    }

    public void toggleComplete() {
        this.completed = !this.completed;
        this.updatedAt = LocalDateTime.now();
    }

    public void assignToList(ReminderList list) {
        this.reminderList = list;
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/reminder.ts
export enum Priority {
  NONE = "NONE",
  LOW = "LOW",
  MEDIUM = "MEDIUM",
  HIGH = "HIGH",
}

export interface Reminder {
  id: number;
  title: string;
  notes: string | null;
  url: string | null;
  remindAt: string | null;
  priority: Priority;
  flagged: boolean;
  completed: boolean;
  sortOrder: number;
  listId: number | null;
  listName: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface ReminderRequest {
  title: string;
  notes?: string;
  url?: string;
  remindAt?: string;
  priority?: Priority;
  flagged?: boolean;
  listId?: number;
}

export interface ReminderList {
  id: number;
  name: string;
  color: string;
  icon: string | null;
  sortOrder: number;
  deletable: boolean;
  reminderCount: number;
  createdAt: string;
  updatedAt: string;
}
